lonestar/eda/cpu/SPRoute/maze_finegrain.h

```diff
@@ -2338,7 +2335,7 @@ void mazeRouteMSMD_finegrain_doall(int iter, int expand, float costHeight,
                         galois::runtime::acquire(&data[grid],
                     galois::MethodFlag::WRITE);*/
 
-                    int preX, preY;
+                    int preX = curX, preY = curY;
```

Comment:
Already merged, but comment here:

Before the change,  I believe it was possible for preX and preY to remain uninitialized even after the following if condition if (1) curr_d1 != 0 AND (2) HV[curY][curX] was 0.

Does this code rely on preX/preY being uninitialized anywhere @Michael-JY-He?

Reply:
Relying on an uninitialized value is undefined behavior, so I hope not. In the case of `preX` and `preY` though, it's not possible. The else clause below used to cover that. That said, I wasn't able to confirm that `tmp`, `tmp_cost`, `tmp_grid`, `tmpX`, and `tmpY` are always initialized. It would be nice to know.

Reply:
I also hope there was no reliance on that.

In the old code it was possible for it to be uninitialized if you entered the first if but not the second if (thereby ignoring the else clause completely).

Reply:
The new code looks good to me. But I still don't understand why the old code is uninitialized? Everybranch will initialize preX and preY right?

Reply:
There are 3 possibilites:

1) curr_d1 != 0 AND HV[curY][curX] != 0
2) curr_d1 != 0 AND HV[curY][curX] **== 0**
3) curr_d1 == 0

(1) and (3) are fine. (2) is where preX and preY will not be initialized if I understand correctly.

There may be an invariant I'm not aware of that prevents (2) from happening.

Reply:
Wait, never mind, the code preview was hiding the "else" that handles case (2) from me. My bad.